test_project/test_app/tests/test_storage.py

```diff
@@ -4,12 +4,80 @@
 
 from test_app.tests.base import AutocompleterTestCase
 from test_app.models import Stock, Indicator
-from test_app.autocompleters import StockAutocompleteProvider, CalcAutocompleteProvider
+from test_app.autocompleters import CalcAutocompleteProvider, FacetedStockAutocompleteProvider, StockAutocompleteProvider
 from test_app import calc_info
-from autocompleter import Autocompleter, registry, signal_registry
+from autocompleter import base, Autocompleter, registry, signal_registry
 from autocompleter import settings as auto_settings
 
 
+class FacetedStoringAndRemovingTestCase(AutocompleterTestCase):
+    fixtures = ['stock_test_data_small.json']
+
+    def test_store_facet_data(self):
+        """
+        Storing saves facet data
+        """
+        aapl = Stock.objects.get(symbol='AAPL')
+        provider = FacetedStockAutocompleteProvider(aapl)
+        provider.store()
+
+        provider_name = provider.get_provider_name()
+        # the FacetedStockAutocompleteProvider get_facets is set to ['sector']
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Technology',)
+        keys = self.redis.keys(facet_set_name)
```

Comment:
i think it would be better to know assume the key exists and do an scard on the set to see how many members it has .. if the scard fails then you know the key is wrong...